pkg/controller-gen/generators/type_go.go

```diff
@@ -360,6 +360,7 @@ func (a *{{.lowerName}}StatusHandler) sync(key string, obj *{{.version}}.{{.type
 	}
 	if !equality.Semantic.DeepEqual(status, newStatus) {
 		var newErr error
+		newStatus.Conditions = obj.Status.Conditions
```

Comment:
pickup the results of the `a.condition.SetError` operations from above

Reply:
Hmm, it looks likes newStatus never escapes this scope.

Reply:
it's copied to `obj.Status` and then persisted via `UpdateStatus` on `obj`